### PR TITLE
Added curve in/out (tangent) handle mirroring

### PR DESCRIPTION
**Pull request by sketchyfun**
(Sorry, re-created this pull request because I well and truly destroyed my last branch trying to squash commits...)

Added a mirroring option to the path editing tool bar for manipulating curve in/out handles (also known as tangent handles), which makes creating curved paths a lot easier. I've added this functionality for both 2D and 3D curves. If you feel the way I've implemented is wrong or could be improved, please let me know :)

2d

3d

### CODE REVIEW

**Comment by sketchyfun** [comment]
Ugh, here we go again. Tried to fix a merge conflict that popped up, and after fixing that, I've apparently brought everyone elses commits into this pull request. Tried reverting it, but that went wrong, so had to revert the revert! This is embarrassing, if anyone can help me figure out what the hell I'm doing wrong, it would be much appreciated :/

**Comment by QbieShay** [comment]
Hi, not 100% sure but you might try this:
- Save your changes in another branch
- force pull godot master on your master
- cherry pick your 2 commits
-rebase squash them

File: editor/plugins/path_editor_plugin.h [hunk 1]
```diff
@@ -78,6 +84,7 @@ class PathEditorPlugin : public EditorPlugin {
 	static PathEditorPlugin *singleton;
 	Ref<SpatialMaterial> path_material;
 	Ref<SpatialMaterial> path_thin_material;
+	CheckBox *mirror_handles;
```
Comment (bojidar-bg) [inline editor/plugins/path_editor_plugin.h]: You aren't using this anywhere else.

**Review by bojidar-bg — APPROVED** [review]
Code looks good overall.

When making changes, you can use `git add` then `git commit --amend` in order to change the existing commit, then use `git push --force` to push it to your branch (since it would offer making a merge between the local and remote branches otherwise).

File: editor/plugins/path_editor_plugin.h (hunk 1 above)
Comment (sketchyfun) [reply editor/plugins/path_editor_plugin.h]: oops, a left over relic, I'll remove that now. Thanks for the git advice too, I'll try it now

**Comment by bojidar-bg** [comment]
Almost there, try to amend and repush again, since you changed the message of the whole commit now.

**Comment by sketchyfun** [comment]
Ah sorry, fixed now :)

**Comment by reduz** [comment]
Please do not add such large labels on the top bar, use an icon or a menu. Alternatively, when we rewrite the property editor the following weeks, may be moved there.

**Comment by sketchyfun** [comment]
I've moved the handle options into an Options menu that appears next to the buttons instead

image

**Comment by mhilbrunner** [comment]
Thanks for your contribution :)

**Pull request merged**